### DISCUSSION TITLE
Help to add button in TinyMCE 6

### DISCUSSION THREAD

**Discussion by AndyCMont** [tinymce/tinymce#8713] [General]
I'm having an issue with the code in the part to get the selected cells.
See the code below:

editor.ui.registry.addButton('sumCells', {
			  icon: 'add',
			  tooltip: 'Sum selected cells',
			  onAction: function () {
				alert('ok 1');
				var selectedCells = editor.selection.getSelectedCells(); /* Error is here */
				alert('ok 2');
				var sum = 0;
				for (var i = 0; i < selectedCells.length; i++) {
				  var cellContent = selectedCells[i].textContent.trim();
				  if (!isNaN(cellContent)) {
					sum += parseFloat(cellContent);
				  }
				}
				editor.notificationManager.open({
				  text: 'Soma ' + sum,
				  type: 'info'
				});
			  }
});

Help me please.

**Comment by lostkeys** [comment]
Hi @AndyCMont. First, [our fiddle tool is a great playground](https://fiddle.tiny.cloud/) for these things, and help us review a complete TinyMCE implementation if you need help.

Regarding adding buttons, [here are a bunch of demos](https://tinymce.github.io/tinymce-demos/index.html) that will get you started, I recommend having a look at [Creating custom toolbar buttons, menus and context toolbars.](https://tinymce.github.io/tinymce-demos/tinymce-basics/custom-toolbar-button-menu-and-context-toolbar.html)

— Fredrik, Product Manager
